homeassistant/components/simplisafe/__init__.py

```diff
@@ -474,6 +474,15 @@ async def async_init(self):
                 _LOGGER.error("Error while fetching initial event: %s", err)
                 self.initial_event_to_use[system.system_id] = {}
 
+            # The initial update doesn't process notifications because it is likely that
+            # they will trigger SIMPLISAFE_NOTIFICATION events before other critical
+            # components – such as automation – are ready. Therefore, we schedule an
+            # update of notifications as soon as HASS is fully started:
+            self._hass.bus.async_listen_once(
+                EVENT_HOMEASSISTANT_START,
```

Comment:
Listening for `EVENT_HOMEASSISTANT_START` isn't quite doing the trick; every now and again, I'll see logs like this:

```txt
2020-04-14 21:46:12 DEBUG (SyncWorker_38) [homeassistant.core] Bus:Handling <Event SIMPLISAFE_NOTIFICATION[L]: category=alarm, code=1000, message=Keypad Panic "Kitchen" triggered, timestamp=2020-04-13T21:13:07-06:00>
2020-04-14 21:46:12 INFO (MainThread) [homeassistant.components.automation] Initialized trigger Notify on SimpliSafe events
2020-04-14 21:46:12 DEBUG (SyncWorker_38) [homeassistant.core] Bus:Handling <Event SIMPLISAFE_NOTIFICATION[L]: category=alarm, code=1001, message=Alarm Cancelled by Master, timestamp=2020-04-13T21:13:18-06:00>
```

In this case, only the second event is captured by my automation; the first event occurs before automation triggers are set up.

Reply:
Why are the events logged in the thread pool? That looks like a possible bug. We're using the async api to fire the events so they should be logged in the main thread if things are working as expected. Is the websocket not async?